magento/magento2
# Throw an exception when asked to translate nothing

### Description
Sometimes developers attempt to translate an empty string. When running `bin/magento i18n:collect-phrases` in these circumstances, an error is shown which prevents the generation of translation files. The templates in question must be fixed before a list of translations can be extracted.

![Screenshot_2022-11-23_11-12-27](https://user-images.githubusercontent.com/334786/203533548-fb0fc2ea-fb40-41f8-a7ec-36bbf6a4bc66.png)

This pull request moves the problem to the be more obvious to module developers. There is currently very little motivation for module developers to fix this problem.

### Manual testing scenarios
1. Ensure Magento is in developer mode with `bin/magento deploy:mode:set developer`
1. Create a template with an empty translation, eg `<?= __('') ?>`
1. Run `bin/magento i18n:collect-phrases`. Witness the error message.
1. View the page with this template. Witness the (lack of) error message.

### Contribution checklist
